### PR TITLE
refactor(ecam/wheel): improve visuals

### PR DESCRIPTION
**Pull request by Benjozork**
Addresses part of #632

## Summary of Changes

 - [x] Refactor code formatting and quality
    - [x] Clean up naming and formatting
    - [x] Make styling and CSS application more consistent
    - [x] Move to a structured view object
    - [x] Extract magic constants
 - [x] Clamp break temperature >0
 - [x] Re-implement / Re-style elements
    - [x] Spoiler indicators
    - [x] Gear door indicators
        - [x] Static
        - [x] In transit
    - [x] Brake temperature indicators
    - [x] Center (ANTI SKID, ALTN BRK, etc.) indicators
    - [x] Auto brake indicators
        - [x] Intensity display
    - [x] LGCIU indicators
        - [x] Display
        - [x] Sensor failure (not displayed yet)
        - [x] Gear unlocked
 - [x] Add HYD elements
    - [x] Display
    - [x] Failure
 - [x] Fix / improve brake temperature colors
 - [x] Add "hottest brake" logic

### Brakes

Before

image

After

image

IRL

image

### Reference screenshots

image

image

#### During cooling

https://www.youtube.com/watch?v=utEPFw0SNMA

#### During landing

https://www.youtube.com/watch?v=g32B8_5TbdQ

## Reference documents

http://easamodul10.blogspot.com/2015/11/a320-series-landing-gear-system.html

Discord username (if different from GitHub): someperson#4953

### CODE REVIEW

**Comment by wpine215** [comment]
Looks like the tire curve lines could be made a bit more shallower and thicker and maybe add some rounding to the edges, just some suggestions when comparing the screenshots to IRL photos.

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_LowerECAM_WHEEL.js [hunk 1]
```diff
@@ -84,31 +95,31 @@ var A320_Neo_LowerECAM_WHEEL;
...
+                if (this.currentBrakeTemps[i] > max) {
                     maxIndex = i;
-                    max = this.CurrentBrakeTemps[i];
+                    max = this.currentBrakeTemps[i];
                 }
             }
 
-            for (var i = 0; i < this.BrakeTempsText.length; i++) {
+            for (let i = 0; i < this.brakeTempsText.length; i++) {
                 if (i === maxIndex && max > 300) {
```
Comment (lukecologne) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_LowerECAM_WHEEL.js:119]:
```suggestion
                if (this.currentBrakeTemps[i] > 300) {
```
According to komp on the discord, every brake temp above 300°C should be amber. The maxIndex can still be used for the "hottest brake arc", if you intend to add that later.

**Review by lukecologne — COMMENTED** [review]
According to @JackZ974, the landing gear triangles shouldn't blink, and there should be an intermediate stage to the gear door symbol. Do you want to take a look at that as well? You seem to be much better at SVGs than me

**Comment by Benjozork** [comment]
Brought positionning of brake indicators closer to the side.

image

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_LowerECAM_WHEEL.js (hunk 1 above)
Comment (Benjozork) [reply A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_LowerECAM_WHEEL.js:119]: I will make sure both of those are implemented.

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_LowerECAM_WHEEL.html [hunk 2]
```diff
@@ -1,4 +1,4 @@
-<link rel="stylesheet" href="./A320_Neo_LowerECAM_WHEEL.css" />
+<link rel="stylesheet" href="./A320_Neo_LowerECAM_WHEEL.css?v=96922225555111s00001110000" />
```
Comment (lukecologne) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_LowerECAM_WHEEL.html:1]: Forgot the cache-breaker. Also shouldn't be necessary with the new devtools version

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_LowerECAM_WHEEL.html [hunk 3]
```diff
@@ -199,54 +222,57 @@
...
+<!--            <path id="path1349-54" d="m71.702583 417.84819h4.729295" style="fill:none;stroke-linecap:round;stroke-width:1.17779px;stroke:#00c800"/>-->
+<!--            <path id="path1349-5-0" d="m71.692161 419.37114h4.729295" style="fill:none;stroke-linecap:round;stroke-width:1.17779px;stroke:#00c800"/>-->
+<!--            <path id="path1349-5-6-26" d="m71.660593 420.91793h4.729295" style="fill:none;stroke-linecap:round;stroke-width:1.17779px;stroke:#00c800"/>-->
+<!--            <path id="path1349-5-6-8-88" d="m71.629026 422.43317h4.729295" style="fill:none;stroke-linecap:round;stroke-width:1.17779px;stroke:#00c800"/>-->
+<!--        </g>-->
     </svg>
 </script>
-<script type="text/html" import-script="/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_LowerECAM_WHEEL.js"></script>
\ No newline at end of file
+<script type="text/html" import-script="/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_LowerECAM_WHEEL.js?v=5551112222"></script>
```
Comment (lukecologne) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_LowerECAM_WHEEL.html:278]: Here as well

**Comment by Benjozork** [comment]
Improved thickness of arches.

image

**Comment by wpine215** [comment]
Looks like the main tires aren't aligned with the landing gear indicators as they should be. You're probably aware of it but wanted to point it out just in case.

**Comment by Benjozork** [comment]
Some progress (ignore the top part 😆)

image

**Comment by Benjozork** [comment]
Added logic for "hottest brake" highlight

image

**Comment by Benjozork** [comment]
Improved appearance of LGCIU indicators (sides only for now !)

image

**Comment by wpine215** [comment]
Brake temperature values seem too low for an RTO performed at 100kts. Furthermore, even after the plane comes to a stop/slows down, the brake temperatures should continue to rise (although slower), as heat from the brakes is transferred to the sensors. Cooling behavior/trend seems to be implemented correctly, however.
FlightSimulator_x9BKTZRZU1

Hydraulics still show as inactive (at least for spoilers) even after 1st engine start.
FlightSimulator_TqNn3SrHVo

Not sure if the green color on one of the anti-skid system borders is intentional, don't see it in the reference pic above.
FlightSimulator_O5AclykGd8

Speedbrake extension should only actuate the middle three spoilers, and aileron movement should only actuate the outer four spoilers - all five are activated only during RTO or landing with spoilers armed.
FlightSimulator_dPERawjOGq
FlightSimulator_mYGchZCQik

Tire SVG curves have flat ends (and sharper curve near the ends), instead of a shallower curve overall, and rounded ends. Try using stroke-linecap="round" to fix the flat ends, possibly?
hkjecI6IPW

**Comment by Benjozork** [comment]
> Brake temperature values seem too low for an RTO performed at 100kts. Furthermore, even after the plane comes to a stop/slows down, the brake temperatures should continue to rise (although slower), as heat from the brakes is transferred to the sensors. Cooling behavior/trend seems to be implemented correctly, however.

I think I will revert the brake temperature changes introduced here and open a separate PR to fix it.

> Hydraulics still show as inactive (at least for spoilers) even after 1st engine start.

Will look into this.

**Comment by wpine215** [comment]
Also, looking at the FSLabs A320, the NORM BRK (in amber), ALTN BRK (<-- ACCU ONLY) and AUTO BRK (in amber) messages are also displayed when N/W STRG switch is set to off, or hydraulics are unavailable. Also, instead of numbers like the F/CTL ECAM when hydraulics are off, the spoilers display amber X's in their place:

Prepar3D_t1wGrxZz5P

Prepar3D_njIi78AOAF

**Comment by Benjozork** [comment]
@wpine215 see IRL reference



Which one should we trust in this case ?

As for NORM BRK showing the ASKID/NW STEERING is off, I will get back to you about that.

**Comment by wpine215** [comment]
Ah, you're right, I didn't realize. The NEO's do have the numbers on the spoilers without hydraulics.

Also, another thing: if the A/SKID & N/W STRG switch is set to off, not only will the center text display, but the temperature readings will be replaced with XX's:
Prepar3D_mILl5hUCA7

**Comment by Benjozork** [comment]
@wpine215 can you show a screenshot with both engines running ?

There are some differences in the pictures you sent with ASKID / NW STEERING off - some show brake temp., some don't.

**Comment by wpine215** [comment]
No engines started, A-SKD & N/W STRG ON:
Prepar3D_oXM12tcuQr

No engines started, A-SKD & N/W STRG OFF:
Prepar3D_SD5Pmo6KCT

Immediately following first engine start, A-SKD & N/W STRG ON:
Prepar3D_PPeimEaCGa

Immediately following first engine start, A-SKD & N/W STRG OFF:
Prepar3D_DKnWoDVYm9

Immediately following second engine start, A-SKD & N/W STRG ON:
Prepar3D_qABgLyMEAY

Immediately following second engine start, A-SKD & N/W STRG OFF
Prepar3D_RZIXriwAUk
:

**Comment by lawfulrocky** [comment]
Name: unlawfulrocky#4343
Date of testing: Current date 09/6/2020
Version of the sim: lastest dev verison
PR Tested: #786

Testing technique: Description of how you tested the Pull Request
Steps you've made to test the Pull Request:
1. Did two abort take offs
2. Did two landings
3. let it sit still to cool the brakes
4. Turn off and on antiskid
5. Put landing gear up and down

Overall Rating: Works great
Conclusions of the testing: all icon works like it should. Brakes heat up when you stop from certain speed and shows on the ecam. all messages shows up as it should and brakes coold down after a certain time.

**Comment by danbw9** [comment]
Name: danbw9
Date of testing: 10/07/2020
Version of the sim: Latest Dev
PR Tested: #786

Testing technique: Description of how you tested the Pull Request
Steps you've made to test the Pull Request:

2 RTOs
2 Landings 1 with LOW 1 with MED
Checked the rate of cooling of the brakes
Turn off and on antiskid and cross check indications
Checked the effect of landing gear being retracted late after take off to see faster cooling of the temps.
Overall Rating: Works as it should.
Conclusions of the testing: Very acceptable! It will be great when the HYD system is simulated and we can link the two systems together to provide greater accuracy,

**Review by wpine215 — APPROVED** [review]

**Pull request merged**